autoware_adapi_v1_msgs/perception/msg/ObjectShape.msg

```diff
@@ -0,0 +1,8 @@
+uint8 UNKNOWN=0
+uint8 BOX=1
+uint8 CYLINDER=2
+uint8 PRISM=3
+
+uint8 type
+geometry_msgs/Vector3 dimensions
+geometry_msgs/Polygon footprint
```

Comment:
I think it's better to add comment documentations for how `dimensions` and `footprint` are handled when different types are chosen.

Reply:
Something similar to https://github.com/ros2/common_interfaces/blob/humble/shape_msgs/msg/SolidPrimitive.msg

Reply:
What about directly using the `SolidPrimitive.msg` message instead of added `ObjectShape.msg`?

Reply:
@xmfcx Actually for this message, I cloned from the autoware perception message
It is the same message type
https://github.com/tier4/autoware_auto_msgs/blob/tier4/main/autoware_auto_perception_msgs/msg/Shape.idl


> What about directly using the SolidPrimitive.msg message instead of added ObjectShape.msg?

@yukkysaito @isamu-takagi How do you think?

Reply:
Oh, we have contributed the missing features to the https://github.com/ros2/common_interfaces/blob/humble/shape_msgs/msg/SolidPrimitive.msg from Autoware.Auto. We were going to remove existing Shape.idl anyways. It's better to use the standard message.

Reply:
I see, so autoware.universe planning to remove the shape.

I that's the case, I'm okay to use the `SolidPrimitive.msg`

Reply:
>I see, so autoware.universe planning to remove the shape.

This should be discussed a bit more.
Autoware's current Shape.idl is focused on the information needed to communicate in Perception-Planning. The interface should be limited to only the necessary information.
However, I think SolidPrimitive.msg is too expressive for general purpose.
Then, even when a Shape contains information that is not used in Autoware, it should be handled in a way that it works properly in Planning.

However, I am not opposed to using SolidPrimitive as an API.

Reply:
Ya, sorry that should not be discuss at here.
For the api, we will use the `SolidPrimitive.msg`

Reply:
@yukkysaito @isamu-takagi @xmfcx 
Sorry for the long delay
I changed the shape to SolidPrimitive.msg at 7187d30
Please approve if it looks good for you